// @puckeditor/core/dist/chunk-4US666EQ.mjs
import {
  useAlignOptions
} from "./chunk-M5GPDW62.mjs";
import {
  useHeadingOptions
} from "./chunk-3RQGAM3Z.mjs";
import {
  useListOptions
} from "./chunk-XIGSFEU7.mjs";
import {
  Action,
  AlignCenter,
  AlignJustify,
  AlignLeft,
  AlignRight,
  Bold,
  Code,
  ControlContext,
  Heading,
  IconButton,
  Italic,
  List,
  ListOrdered,
  Minus,
  Quote,
  SelectControl,
  SquareCode,
  Strikethrough,
  Underline,
  useControlContext
} from "./chunk-XQHCQEAF.mjs";
import {
  get_class_name_factory_default
} from "./chunk-Y2EFNT5P.mjs";
import {
  init_react_import
} from "./chunk-M6W7YEVX.mjs";

// components/RichTextMenu/inner.tsx
init_react_import();

// css-module:/home/runner/work/puck/puck/packages/core/components/RichTextMenu/styles.module.css#css-module
init_react_import();
var styles_module_default = { "RichTextMenu": "_RichTextMenu_1ve2j_1", "RichTextMenu--form": "_RichTextMenu--form_1ve2j_7", "RichTextMenu-group": "_RichTextMenu-group_1ve2j_21", "RichTextMenu--inline": "_RichTextMenu--inline_1ve2j_39" };

// components/RichTextMenu/inner.tsx
import { useMemo } from "react";

// components/RichTextMenu/controls/index.ts
init_react_import();

// components/RichTextMenu/controls/AlignLeft.tsx
init_react_import();

// components/RichTextMenu/components/Control/index.tsx
init_react_import();

// css-module:/home/runner/work/puck/puck/packages/core/components/RichTextMenu/components/Control/styles.module.css#css-module
init_react_import();
var styles_module_default2 = { "Control": "_Control_id4pm_1", "Control--inline": "_Control--inline_id4pm_6" };

// components/RichTextMenu/components/Control/index.tsx
import { jsx } from "react/jsx-runtime";
var getClassName = get_class_name_factory_default("Control", styles_module_default2);
function Control({
  icon,
  disabled,
  active,
  onClick,
  title
}) {
  const { inline } = useControlContext();
  if (inline) {
    return /* @__PURE__ */ jsx("span", { className: getClassName({ inline: true }), children: /* @__PURE__ */ jsx(
      Action,
      {
        onClick,
        disabled,
        active,
        label: title,
        children: icon
      }
    ) });
  }
  return /* @__PURE__ */ jsx("span", { className: getClassName(), children: /* @__PURE__ */ jsx(
    IconButton,
    {
      onClick,
      disabled,
      active,
      title,
      children: icon
    }
  ) });
}

// components/RichTextMenu/controls/AlignLeft.tsx
import { jsx as jsx2 } from "react/jsx-runtime";
function AlignLeft2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx2(
    Control,
    {
      icon: /* @__PURE__ */ jsx2(AlignLeft, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().setTextAlign("left").run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canAlignLeft),
      active: editorState == null ? void 0 : editorState.isAlignLeft,
      title: "Align left"
    }
  );
}

// components/RichTextMenu/controls/AlignCenter.tsx
init_react_import();
import { jsx as jsx3 } from "react/jsx-runtime";
function AlignCenter2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx3(
    Control,
    {
      icon: /* @__PURE__ */ jsx3(AlignCenter, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().setTextAlign("center").run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canAlignCenter),
      active: editorState == null ? void 0 : editorState.isAlignCenter,
      title: "Align center"
    }
  );
}

// components/RichTextMenu/controls/AlignRight.tsx
init_react_import();
import { jsx as jsx4 } from "react/jsx-runtime";
function AlignRight2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx4(
    Control,
    {
      icon: /* @__PURE__ */ jsx4(AlignRight, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().setTextAlign("right").run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canAlignRight),
      active: editorState == null ? void 0 : editorState.isAlignRight,
      title: "Align right"
    }
  );
}

// components/RichTextMenu/controls/AlignJustify.tsx
init_react_import();
import { jsx as jsx5 } from "react/jsx-runtime";
function AlignJustify2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx5(
    Control,
    {
      icon: /* @__PURE__ */ jsx5(AlignJustify, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().setTextAlign("justify").run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canAlignJustify),
      active: editorState == null ? void 0 : editorState.isAlignJustify,
      title: "Justify"
    }
  );
}

// components/RichTextMenu/controls/AlignSelect/index.tsx
init_react_import();
import { lazy, Suspense } from "react";

// components/RichTextMenu/controls/AlignSelect/fallback.tsx
init_react_import();
import { jsx as jsx6 } from "react/jsx-runtime";
function AlignSelectFallback() {
  const ctx = useControlContext();
  const alignOptions = useAlignOptions(ctx.options);
  return /* @__PURE__ */ jsx6(
    SelectControl,
    {
      options: alignOptions,
      onChange: () => {
      },
      value: "left",
      defaultValue: "left",
      renderDefaultIcon: AlignLeft
    }
  );
}

// components/RichTextMenu/controls/AlignSelect/index.tsx
import { jsx as jsx7 } from "react/jsx-runtime";
var AlignSelectLoaded = lazy(
  () => import("./loaded-QH5RKCGC.mjs").then((m) => ({
    default: m.AlignSelectLoaded
  }))
);
var AlignSelect = () => /* @__PURE__ */ jsx7(Suspense, { fallback: /* @__PURE__ */ jsx7(AlignSelectFallback, {}), children: /* @__PURE__ */ jsx7(AlignSelectLoaded, {}) });

// components/RichTextMenu/controls/Bold.tsx
init_react_import();
import { jsx as jsx8 } from "react/jsx-runtime";
function Bold2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx8(
    Control,
    {
      icon: /* @__PURE__ */ jsx8(Bold, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleBold().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canBold),
      active: editorState == null ? void 0 : editorState.isBold,
      title: "Bold"
    }
  );
}

// components/RichTextMenu/controls/Italic.tsx
init_react_import();
import { jsx as jsx9 } from "react/jsx-runtime";
function Italic2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx9(
    Control,
    {
      icon: /* @__PURE__ */ jsx9(Italic, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleItalic().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canItalic),
      active: editorState == null ? void 0 : editorState.isItalic,
      title: "Italic"
    }
  );
}

// components/RichTextMenu/controls/Underline.tsx
init_react_import();
import { jsx as jsx10 } from "react/jsx-runtime";
function Underline2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx10(
    Control,
    {
      icon: /* @__PURE__ */ jsx10(Underline, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleUnderline().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canUnderline),
      active: editorState == null ? void 0 : editorState.isUnderline,
      title: "Underline"
    }
  );
}

// components/RichTextMenu/controls/Strikethrough.tsx
init_react_import();
import { jsx as jsx11 } from "react/jsx-runtime";
function Strikethrough2() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx11(
    Control,
    {
      icon: /* @__PURE__ */ jsx11(Strikethrough, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleStrike().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canStrike),
      active: editorState == null ? void 0 : editorState.isStrike,
      title: "Strikethrough"
    }
  );
}

// components/RichTextMenu/controls/InlineCode.tsx
init_react_import();
import { jsx as jsx12 } from "react/jsx-runtime";
function InlineCode() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx12(
    Control,
    {
      icon: /* @__PURE__ */ jsx12(Code, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleCode().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canInlineCode),
      active: editorState == null ? void 0 : editorState.isInlineCode,
      title: "Inline code"
    }
  );
}

// components/RichTextMenu/controls/BulletList.tsx
init_react_import();
import { jsx as jsx13 } from "react/jsx-runtime";
function BulletList() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx13(
    Control,
    {
      icon: /* @__PURE__ */ jsx13(List, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleBulletList().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canBulletList),
      active: editorState == null ? void 0 : editorState.isBulletList,
      title: "Bullet list"
    }
  );
}

// components/RichTextMenu/controls/OrderedList.tsx
init_react_import();
import { jsx as jsx14 } from "react/jsx-runtime";
function OrderedList() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx14(
    Control,
    {
      icon: /* @__PURE__ */ jsx14(ListOrdered, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleOrderedList().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canOrderedList),
      active: editorState == null ? void 0 : editorState.isOrderedList,
      title: "Ordered list"
    }
  );
}

// components/RichTextMenu/controls/CodeBlock.tsx
init_react_import();
import { jsx as jsx15 } from "react/jsx-runtime";
function CodeBlock() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx15(
    Control,
    {
      icon: /* @__PURE__ */ jsx15(SquareCode, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleCodeBlock().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canCodeBlock),
      active: editorState == null ? void 0 : editorState.isCodeBlock,
      title: "Code block"
    }
  );
}

// components/RichTextMenu/controls/Blockquote.tsx
init_react_import();
import { jsx as jsx16 } from "react/jsx-runtime";
function Blockquote() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx16(
    Control,
    {
      icon: /* @__PURE__ */ jsx16(Quote, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().toggleBlockquote().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canBlockquote),
      active: editorState == null ? void 0 : editorState.isBlockquote,
      title: "Blockquote"
    }
  );
}

// components/RichTextMenu/controls/HorizontalRule.tsx
init_react_import();
import { jsx as jsx17 } from "react/jsx-runtime";
function HorizontalRule() {
  const { editor, editorState } = useControlContext();
  return /* @__PURE__ */ jsx17(
    Control,
    {
      icon: /* @__PURE__ */ jsx17(Minus, {}),
      onClick: (e) => {
        e.stopPropagation();
        editor == null ? void 0 : editor.chain().focus().setHorizontalRule().run();
      },
      disabled: !(editorState == null ? void 0 : editorState.canHorizontalRule),
      title: "Horizontal rule"
    }
  );
}

// components/RichTextMenu/controls/HeadingSelect/index.tsx
init_react_import();
import { lazy as lazy2, Suspense as Suspense2 } from "react";

// components/RichTextMenu/controls/HeadingSelect/fallback.tsx
init_react_import();
import { jsx as jsx18 } from "react/jsx-runtime";
function HeadingSelectFallback() {
  const ctx = useControlContext();
  const headingOptions = useHeadingOptions(ctx.options);
  return /* @__PURE__ */ jsx18(
    SelectControl,
    {
      options: headingOptions,
      onChange: () => {
      },
      value: "p",
      defaultValue: "p",
      renderDefaultIcon: Heading
    }
  );
}

// components/RichTextMenu/controls/HeadingSelect/index.tsx
import { jsx as jsx19 } from "react/jsx-runtime";
var HeadingSelectLoaded = lazy2(
  () => import("./loaded-LAKI4KDY.mjs").then((m) => ({
    default: m.HeadingSelectLoaded
  }))
);
var HeadingSelect = () => /* @__PURE__ */ jsx19(Suspense2, { fallback: /* @__PURE__ */ jsx19(HeadingSelectFallback, {}), children: /* @__PURE__ */ jsx19(HeadingSelectLoaded, {}) });

// components/RichTextMenu/controls/ListSelect/index.tsx
init_react_import();
import { lazy as lazy3, Suspense as Suspense3 } from "react";

// components/RichTextMenu/controls/ListSelect/fallback.tsx
init_react_import();
import { jsx as jsx20 } from "react/jsx-runtime";
function ListSelectFallback() {
  const ctx = useControlContext();
  const listOptions = useListOptions(ctx.options);
  return /* @__PURE__ */ jsx20(
    SelectControl,
    {
      options: listOptions,
      onChange: () => {
      },
      value: "p",
      defaultValue: "p",
      renderDefaultIcon: List
    }
  );
}

// components/RichTextMenu/controls/ListSelect/index.tsx
import { jsx as jsx21 } from "react/jsx-runtime";
var ListSelectLoaded = lazy3(
  () => import("./loaded-PILFUJGA.mjs").then((m) => ({
    default: m.ListSelectLoaded
  }))
);
var ListSelect = () => /* @__PURE__ */ jsx21(Suspense3, { fallback: /* @__PURE__ */ jsx21(ListSelectFallback, {}), children: /* @__PURE__ */ jsx21(ListSelectLoaded, {}) });

// components/RichTextMenu/inner.tsx
import { jsx as jsx22, jsxs } from "react/jsx-runtime";
var getClassName2 = get_class_name_factory_default("RichTextMenu", styles_module_default);
var DefaultMenu = ({ children }) => {
  return /* @__PURE__ */ jsx22(RichTextMenu, { children });
};
var RichTextMenu = ({ children }) => {
  const { inline } = useControlContext();
  return /* @__PURE__ */ jsx22("div", { className: getClassName2({ inline, form: !inline }), "data-puck-rte-menu": true, children });
};
var Group = ({ children }) => {
  return /* @__PURE__ */ jsx22("div", { className: getClassName2("group"), children });
};
RichTextMenu.Group = Group;
RichTextMenu.Control = Control;
RichTextMenu.AlignCenter = AlignCenter2;
RichTextMenu.AlignJustify = AlignJustify2;
RichTextMenu.AlignLeft = AlignLeft2;
RichTextMenu.AlignRight = AlignRight2;
RichTextMenu.AlignSelect = AlignSelect;
RichTextMenu.Blockquote = Blockquote;
RichTextMenu.Bold = Bold2;
RichTextMenu.BulletList = BulletList;
RichTextMenu.CodeBlock = CodeBlock;
RichTextMenu.HeadingSelect = HeadingSelect;
RichTextMenu.HorizontalRule = HorizontalRule;
RichTextMenu.InlineCode = InlineCode;
RichTextMenu.Italic = Italic2;
RichTextMenu.ListSelect = ListSelect;
RichTextMenu.OrderedList = OrderedList;
RichTextMenu.Strikethrough = Strikethrough2;
RichTextMenu.Underline = Underline2;
var LoadedRichTextMenuInner = ({
  editor = null,
  editorState = null,
  field,
  readOnly,
  inline
}) => {
  const { renderMenu, renderInlineMenu } = field;
  const InlineMenu = useMemo(
    () => renderInlineMenu || DefaultMenu,
    [renderInlineMenu]
  );
  const Menu = useMemo(() => renderMenu || DefaultMenu, [renderMenu]);
  return /* @__PURE__ */ jsx22(
    ControlContext.Provider,
    {
      value: { editor, editorState, inline, options: field.options, readOnly },
      children: inline ? /* @__PURE__ */ jsx22(
        InlineMenu,
        {
          editor,
          editorState,
          readOnly,
          children: /* @__PURE__ */ jsxs(Group, { children: [
            /* @__PURE__ */ jsx22(Bold2, {}),
            /* @__PURE__ */ jsx22(Italic2, {}),
            /* @__PURE__ */ jsx22(Underline2, {})
          ] })
        }
      ) : /* @__PURE__ */ jsxs(Menu, { editor, editorState, readOnly, children: [
        /* @__PURE__ */ jsxs(Group, { children: [
          /* @__PURE__ */ jsx22(HeadingSelect, {}),
          /* @__PURE__ */ jsx22(ListSelect, {})
        ] }),
        /* @__PURE__ */ jsxs(Group, { children: [
          /* @__PURE__ */ jsx22(Bold2, {}),
          /* @__PURE__ */ jsx22(Italic2, {}),
          /* @__PURE__ */ jsx22(Underline2, {})
        ] }),
        /* @__PURE__ */ jsx22(Group, { children: /* @__PURE__ */ jsx22(AlignSelect, {}) })
      ] })
    }
  );
};

export {
  RichTextMenu,
  LoadedRichTextMenuInner
};
